huaweicloud/data_source_huaweicloud_gaussdb_mysql_instances.go

```diff
@@ -0,0 +1,314 @@
+						"datastore": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"engine": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"version": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"backup_strategy": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"start_time": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"keep_days": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"read_replicas": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"flavor": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"nodes": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"name": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"type": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"status": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"private_read_ip": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"availability_zone": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceGaussDBMysqlInstancesRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	region := GetRegion(d, config)
+	client, err := config.gaussdbV3Client(region)
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud GaussDB client: %s", err)
+	}
+
+	listOpts := instances.ListTaurusDBInstanceOpts{
+		Name:     d.Get("name").(string),
+		VpcId:    d.Get("vpc_id").(string),
+		SubnetId: d.Get("subnet_id").(string),
+	}
+
+	pages, err := instances.List(client, listOpts).AllPages()
+	if err != nil {
+		return err
+	}
+
+	allInstances, err := instances.ExtractTaurusDBInstances(pages)
+
+	if err != nil {
+		return fmt.Errorf("Unable to retrieve instances: %s", err)
+	}
+
+	if allInstances.TotalCount < 1 {
+		return fmt.Errorf("Your query returned no results. " +
+			"Please change your search criteria and try again.")
+	}
+
+	var instancesToSet []map[string]interface{}
+	var instancesIds []string
+
+	for _, instanceInAll := range allInstances.Instances {
+		instanceID := instanceInAll.Id
+		instancesIds = append(instancesIds, instanceID)
+		instance, err := instances.Get(client, instanceID).Extract()
```

Comment:
why we get the DB details again?

Reply:
DB detail returns more information than list instances, such as nodes.
You can compare the API docs.